docs/api/v1/vm.md

```diff
@@ -0,0 +1,57 @@
+## /vm
+
+```
+vm = {
+  id: int,
+  cores: int,
+  ram: Bytes (int),
+  customer_id: int,
+  cputime_limit: int,
+  uuid: UUID (String),
+  state: {
+    id: StateID (int),
+    name: String,
+    description: String,
+  },
+  node_method: {
+    id: int,
+    virt_method: {
+      id: int,
+      name: string,
+    },
+  },
+}
+```
+
+| Verb | Notes |
+|------|-------|
+| GET  | Returns a list of `vm` objects. |
+| POST | Adds a new vm. `vm.id` is ignored. The UUID will be auto generated if you don't provide it |
+| DELETE | Removes a vm. Only `vm.id` has to be set. |
+| UPDATE | Updates a vm. `vm.id` is required, additional values set will be updated. |
+
+### Response
+
+All verbs except DELETE return the new/current vm object(s), DELETE returns the deleted object(s).
+
+### Errors
+
+| Error ID | Message | Notes |
+|----------|---------|-------|
+| example id | message  | notes |
+
+### Notes
+
+A `POST` call will create a VM and a network interface without a VLAN, but with a free IPv4 and IPv6 address. You might want to create a blockdevice [(/storage](storage.md)) and assign it to the new VM. The `cputime_limit` refers to the [Qemu driver for cgroups cpu controller](https://libvirt.org/cgroups.html). You can set `vm.node_method.id` OR `vm.node_method.virt_method.id`. The first one specifies the hardware node AND the needed hypervisor technology, the second one specifies only the needed hypervisor technology (for example KVM) and the API itself will choose a suitable host system.
```

Comment:
Should a new vm really automatically get a network interface? It might be better if the vm_interface endpoint has an easy way to do this rather than doing it automatically here. We don't create storage so we shouldn't create the network either.